pkg/insight/filestore.go

```diff
@@ -68,6 +71,30 @@ func (s *Store) PutChunk(ctx context.Context, chunk Chunk) error {
 	return s.filestore.PutObject(ctx, path, data)
 }
 
+func (s *Store) LoadChunksCache(cache cache.Cache, projectID, appID string, kind model.InsightMetricsKind, step model.InsightStep, from time.Time, count int) ([]Chunk, error) {
+	var chunks Chunks
+	paths := determineFilePaths(projectID, appID, kind, step, from, count)
+	for _, p := range paths {
+		c, err := cache.Get(p)
+		chunk, ok := c.(Chunk)
+		if err != nil || !ok {
+			return nil, fmt.Errorf("failed to get insight from cache")
+		}
+		chunks = append(chunks, chunk)
+	}
+
+	return chunks, nil
+}
+
+func (s *Store) PutChunksToCache(cache cache.Cache, chunks Chunks) error {
+	var err error
+	for _, c := range chunks {
+		// continue process even if an error occurs.
+		err = cache.Put(c.GetFilePath(), c)
```

Comment:
When the last `Put` is successful, the previous error will be ignored and no error is returned.
So it should be:

```golang
if e := cache.Put(c.GetFilePath(), c); e != nil {
    err = e
}
```

Reply:
Oops, thanks ;  ;